module-gui/gui/widgets/text/core/cursors/TextInLineCursor.cpp

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "TextInLineCursor.hpp"
+#include <Text.hpp>
+#include <log/log.hpp>
+
+#if DEBUG_GUI_TEXT_CURSOR == 1
+#define debug_text_cursor(...) LOG_DEBUG(__VA_ARGS__)
+#else
+#define debug_text_cursor(...)
+#endif
+
+namespace gui
+{
+    TextInLineCursor::TextInLineCursor(gui::Text *parent, unsigned int pos, unsigned int block)
+        : TextCursor(parent, pos, block)
+    {}
+
+    bool TextInLineCursor::checkInLineLeftMove()
+    {
+        return onScreenPosition == 0 && (BlockCursor::getBlockNumber() != 0 || BlockCursor::getPosition() != 0);
+    }
+
+    void TextInLineCursor::moveInLineLeft()
+    {
+        const auto [startDrawBlockNumber, startDrawBlockPosition] = text->lines->drawStartConditions;
+        BlockCursor moveCursor(
+            getDocument(), startDrawBlockPosition, startDrawBlockNumber, text->getTextFormat().getFont());
+        --moveCursor;
+
+        text->lines->drawStartConditions = {moveCursor.getBlockNumber(), moveCursor.getPosition()};
+        text->drawLines();
+    }
+
+    bool TextInLineCursor::checkInLineRightMove()
+    {
+        return onScreenPosition + 1 > text->lines->getLine()->length() &&
```

Comment:
Minor. Could be without extra addition: 
```suggestion
        return onScreenPosition >= text->lines->getLine()->length() &&
```

Reply:
sure